cfme/tests/infrastructure/test_datastore_analysis.py

```diff
@@ -56,41 +57,48 @@ def pytest_generate_tests(metafunc):
     testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope="module")
 
 
-@pytest.fixture(scope='module')
-def datastore(appliance, provider, datastore_type, datastore_name):
-    return appliance.collections.datastores.instantiate(name=datastore_name,
-                                                        provider=provider,
-                                                        type=datastore_type)
-
-
-@pytest.fixture(scope='module')
-def datastores_hosts_setup(provider, datastore):
-    hosts = datastore.hosts.all()
-    for host in hosts:
-        host_data = [data
-                     for data in provider.data.get("hosts", {})
-                     if data.get("name") == host.name]
-        if not host_data:
-            pytest.skip(f"No host data for provider {provider} and datastore {datastore}")
-        host.update_credentials_rest(credentials=host_data[0]['credentials'])
-    else:
-        pytest.skip(f"No hosts attached to the datastore selected for testing: {datastore}")
+@pytest.fixture
+def datastore(temp_appliance_preconfig_funcscope, provider, datastore_type, datastore_name)\
+        -> Datastore:
+    with temp_appliance_preconfig_funcscope as appliance:
+        return appliance.collections.datastores.instantiate(name=datastore_name,
+                                                            provider=provider,
+                                                            type=datastore_type)
+
+
+@pytest.fixture
+def datastores_hosts_setup(setup_provider_temp_appliance, provider, datastore):
+    updated_hosts = []
+    for host in datastore.hosts.all():
+        try:
+            host_data, = [data
+                          for data in provider.data.get("hosts", {})
+                          if data.get("name") == host.name]
+        except ValueError as exc:
+            pytest.skip(f"Data for host {host} in provider {provider} and datastore {datastore} "
+                        f"couldn't be determined: {exc}.")
+        else:
+            host.update_credentials_rest(credentials=host_data['credentials'])
+            updated_hosts.append(host)
+
+    if not updated_hosts:
+        pytest.skip(f"No hosts attached to the datastore {datastore} was selected for testing.")
     yield
-    for host in hosts:
+    for host in updated_hosts:
         host.remove_credentials_rest()
 
 
-@pytest.fixture(scope='function')
+@pytest.fixture()
 def clear_all_tasks(appliance):
     # clear table
     col = appliance.collections.tasks.filter({'tab': 'AllTasks'})
     col.delete_all()
 
 
 @pytest.mark.tier(2)
-def test_run_datastore_analysis(setup_provider, datastore, soft_assert, datastores_hosts_setup,
-                                clear_all_tasks, appliance):
-    """Tests smarthost analysis
+def test_run_datastore_analysis(setup_provider_temp_appliance, datastore, soft_assert,
```

Comment:
Why change this to use temp_appliance? I don't see anything here that would disrupt the appliance itself?

Reply:
I have added a comment above about that. I should have mentioned it before I made this RFR